src/test/java/org/wildfly/security/authz/RoleMappingTest.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2018 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.authz;
+
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Basic String role mapping tests
+ *
+ * @author <a href="mailto:mmazanek@redhat.com">Martin Mazanek</a>
+ */
+public class RoleMappingTest {
+
+    @Test
+    public void testMappedRoles1() {
+        Roles roles = createRoles("foo", "joe");
+
+        Map<String, Set<String>> mappingMap = new HashMap<>();
+        mappingMap.put("foo", createSet("bar", "role"));
+
+        RoleMapper mapper = new MappedRoleMapper.Builder()
+                .setRemoveMapped(true)
+                .setRemoveNonMapped(true)
+                .setRoleMap(mappingMap).build();
+
+        Roles mappedRoles = mapper.mapRoles(roles);
+
+        assertTrue(mappedRoles.contains("bar"));
+        assertTrue(mappedRoles.contains("role"));
+        assertFalse(mappedRoles.contains("foo"));
+        assertFalse(mappedRoles.contains("joe"));
+
+        Iterator<String> iterator = mappedRoles.iterator();
+        int count = 0;
+        while (iterator.hasNext()) {
+            iterator.next();
+            count++;
+        }
+        assertTrue(count == 2);
+    }
+
+    @Test
+    public void testMappedRoles2() {
+        Roles roles = createRoles("foo", "joe");
+
+        Map<String, Set<String>> mappingMap = new HashMap<>();
+        mappingMap.put("foo", createSet("bar", "role"));
```

Comment:
Is it possible to add tests with multiple records in mappingMap? With intersection in keys and values.

Reply:
will do